src/plugins/topic_viewer/TopicViewer.cc

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include "TopicViewer.hh"
+#include <ignition/plugin/Register.hh>
+
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+  class TopicsModel : public QStandardItemModel
+  {
+    /// \brief roles and names of the model
+    public: QHash<int, QByteArray> roleNames() const override
+    {
+      QHash<int, QByteArray> roles;
+      roles[NAME_ROLE] = NAME_KEY;
+      roles[TYPE_ROLE] = TYPE_KEY;
+      roles[TOPIC_ROLE] = TOPIC_KEY;
+      roles[PATH_ROLE] = PATH_KEY;
+      roles[PLOT_ROLE] = PLOT_KEY;
+      return roles;
+    }
+  };
+
+  class TopicViewerPrivate
+  {
+    /// \brief Node for Commincation
+    public: ignition::transport::Node node;
+
+    /// \brief Model to create it from the available topics and messages
+    public: TopicsModel *model;
+  };
+}
+}
+}
+using namespace ignition;
+using namespace gui;
+using namespace plugins;
+
+TopicViewer::TopicViewer() :Plugin(), dataPtr(new TopicViewerPrivate)
+{
+  using namespace google::protobuf;
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_DOUBLE);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_FLOAT);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_INT32);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_INT64);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_UINT32);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_UINT64);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_BOOL);
+
+  this->plottingMode = true;
+
+  this->CreateModel();
+
+  ignition::gui::App()->Engine()->rootContext()->setContextProperty(
+                "TopicsModel", this->dataPtr->model);
+}
+
+TopicViewer::~TopicViewer()
+{
+}
+
+void TopicViewer::LoadConfig(const tinyxml2::XMLElement *)
+{
+  if (this->title.empty())
+    this->title = "Topic Viewer";
+}
+
```

Comment:
```suggestion

//////////////////////////////////////////////////
```
style nitpick: add line segregation between each function